### PR TITLE
fixing debian and arch concurrent multiarch builds

### LINKED ISSUE #9995: minikube_1.16.0-0_amd64.deb: package architecture (arm64) does not match system (amd64)
**Steps to reproduce the issue:** 

1. Download amd64 debian package of latest release: https://github.com/kubernetes/minikube/releases/download/v1.16.0/minikube_1.16.0-0_amd64.deb
2. Inspect the package file
3. It's marked as an arm64 package not an amd64 package
4. It has stray files in it that are clearly not meant to be there

**Full output of failed command:** 
```
$ dpkg --info minikube_1.16.0-0_amd64.deb ; dpkg --contents minikube_1.16.0-0_amd64.deb 
 new Debian package, version 2.0.
 size 20578594 bytes: control archive=389 bytes.
     381 bytes,    11 lines      control              
 Package: minikube
 Version: 1.16.0
 Section: base
 Priority: optional
 Architecture: arm64
 Maintainer: Thomas Strömberg <t+minikube@stromberg.org>
 Description: Minikube
  minikube is a tool that makes it easy to run Kubernetes locally.
  minikube runs a single-node Kubernetes cluster inside a VM on your
  laptop for users looking to try out Kubernetes or develop with it 
  day-to-day.
drwxr-xr-x root/root         0 2020-12-17 18:41 ./
drwxr-xr-x root/root         0 2020-12-17 18:41 ./minikube_deb_template/
drwxr-xr-x root/root         0 2020-12-17 18:41 ./minikube_deb_template/DEBIAN/
-rw-r--r-- root/root       416 2020-12-17 18:41 ./minikube_deb_template/DEBIAN/control
drwxr-xr-x root/root         0 2020-12-17 18:41 ./usr/
drwxr-xr-x root/root         0 2020-12-17 18:41 ./usr/bin/
-rwxr-xr-x root/root  57925077 2020-12-17 18:41 ./usr/bin/minikube
```

Note: `Architecture: arm64` and `./minikube_deb_template/...` in the contents.

However the actual binary contained within _is_ an amd64 one:
```
# extract the contents of the package
$ file minikube
minikube: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, Go BuildID=6LZF01FcZrGBprW2lwYI/gWzE5fRYw4NT_meHguiF/54zHaIwKVnjc2RZCfg6t/T0VXjXYTfvHZFrWQslv4, not stripped
```

### PR DESCRIPTION
**Pull request by Jorropo**
fixes #9995

### CODE REVIEW

**Comment by afbjorklund** [comment]
I'm thinking now might be the time to introduce a variable for it ? :-)

Also need to fix the rpm build, which has same issue as deb build

File: Makefile [hunk 1]
```diff
@@ -467,19 +467,19 @@ out/minikube_$(DEB_VERSION).deb: out/minikube_$(DEB_VERSION)-0_amd64.deb
 	cp $< $@
 
 out/minikube_$(DEB_VERSION)-0_%.deb: out/minikube-linux-%
-	cp -r installers/linux/deb/minikube_deb_template out/minikube_$(DEB_VERSION)
-	chmod 0755 out/minikube_$(DEB_VERSION)/DEBIAN
-	sed -E -i 's/--VERSION--/'$(DEB_VERSION)'/g' out/minikube_$(DEB_VERSION)/DEBIAN/control
-	sed -E -i 's/--ARCH--/'$*'/g' out/minikube_$(DEB_VERSION)/DEBIAN/control
+	cp -r installers/linux/deb/minikube_deb_template out/minikube_$(DEB_VERSION)-$*
```
Comment (afbjorklund) [inline Makefile:470]: Maybe put `minikube_$(DEB_VERSION)-$*` into a variable ?

File: Makefile [hunk 2]
```diff
@@ -467,19 +467,20 @@ out/minikube_$(DEB_VERSION).deb: out/minikube_$(DEB_VERSION)-0_amd64.deb
 	cp $< $@
 
 out/minikube_$(DEB_VERSION)-0_%.deb: out/minikube-linux-%
-	cp -r installers/linux/deb/minikube_deb_template out/minikube_$(DEB_VERSION)
-	chmod 0755 out/minikube_$(DEB_VERSION)/DEBIAN
-	sed -E -i 's/--VERSION--/'$(DEB_VERSION)'/g' out/minikube_$(DEB_VERSION)/DEBIAN/control
-	sed -E -i 's/--ARCH--/'$*'/g' out/minikube_$(DEB_VERSION)/DEBIAN/control
+	PACKAGING_DIRECTORY="out/minikube_$(DEB_VERSION)-$*"
```
Comment (afbjorklund) [inline Makefile:470]: This is make, and not bash, so this doesn't work - the make variable will not be set further down

You can verify, by looking at the generated commands: `make -n out/minikube_1.16.0.deb`

**Review by afbjorklund — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (mgabeler-lee-6rs) [reply Makefile:470]: You can use [pattern-specific variables](https://www.gnu.org/software/make/manual/html_node/Pattern_002dspecific.html#Pattern_002dspecific) to declare this as a "local" variable for this target, e.g.:
```
out/minikube_$(DEB_VERSION)-0_%.deb: PACKAGING_DIRECTORY=out/minikube_$(DEB_VERSION)-$*
out/minikube_$(DEB_VERSION)-0_%.deb: out/minikube-linux-%
```
NB: you _must_ put the variable and the pattern rule on separate lines, can't amend the variable as a third item after the pattern in the same line

Comment (Jorropo) [reply Makefile]: That fixed, now that you said it that obvious, it works now.

File: Makefile [hunk 3]
```diff
@@ -466,35 +466,37 @@ deb_version:
...
+	fakeroot dpkg-deb --build $(PACKAGING_DIRECTORY) $@
+	rm -rf $(PACKAGING_DIRECTORY)
 
 rpm_version:
 	@echo $(RPM_VERSION)
 
 out/minikube-$(RPM_VERSION).rpm: out/minikube-$(RPM_VERSION)-0.x86_64.rpm
 	cp $< $@
 
+out/minikube_$(RPM_VERSION)-0_%.deb: PACKAGING_DIRECTORY=out/minikube_$(RPM_VERSION)-$*
```
Comment (afbjorklund) [inline Makefile:491]: Note that the original said "minikube-$(RPM_VERSION)" (with hyphen, not underscore)

Now we get a race between the deb and the rpm instead, and it still fails to build `make -j`

**Review by afbjorklund — CHANGES_REQUESTED** [review]
Looking good! Almost there, just one little byte left

**Comment by afbjorklund** [comment]
Reproducer:
```console
$ DEB_VERSION=$(make dev_version)
$ RPM_VERSION=$(make rpm_version)
$ make -j   "out/minikube_${DEB_VERSION}-0_amd64.deb"   "out/minikube_${DEB_VERSION}-0_arm64.deb"   "out/minikube-${RPM_VERSION}-0.x86_64.rpm"   "out/minikube-${RPM_VERSION}-0.aarch64.rpm"
cp -r installers/linux/deb/minikube_deb_template out/minikube_1.16.0-amd64
cp -r installers/linux/deb/minikube_deb_template out/minikube_1.16.0-arm64
cp -r installers/linux/rpm/minikube_rpm_template 
cp -r installers/linux/rpm/minikube_rpm_template 
cp: missing destination file operand after 'installers/linux/rpm/minikube_rpm_template'
Try 'cp --help' for more information.
```

File: Makefile (hunk 3 above)
Comment (afbjorklund) [reply Makefile:491]: Need to use two different variables:

```
DEB_PACKAGING_DIRECTORY=out/minikube_$(DEB_VERSION)-$*
RPM_PACKAGING_DIRECTORY=out/minikube-$(RPM_VERSION)-$*
```

Next time, we'll use mktemp instead.

**Review by afbjorklund — APPROVED** [review]

**Comment by medyagh** [comment]
@ilya-zuyev is this trying to fix same problem as  https://github.com/kubernetes/minikube/pull/10004 ?

**Pull request merged**